app/controllers/api/v1/npq_applications_controller.rb

```diff
@@ -20,6 +20,29 @@ def index
         end
       end
 
+      def reject
+        profile = npq_lead_provider.npq_profiles.includes(:user, :npq_course).find(params[:id])
+
+        if profile.update(lead_provider_approval_status: "rejected")
+          render json: NPQApplicationSerializer.new(profile).serializable_hash
+        else
+          render json: { errors: Api::ErrorFactory.new(model: profile).call }, status: :bad_request
+        end
+      end
+
+      def accept
+        profile = npq_lead_provider.npq_profiles.includes(:user, :npq_course).find(params[:id])
+        other_profiles = NPQValidationData.where(profile: (profile.user.npq_profiles - [profile.profile]))
+
+        ActiveRecord::Base.transaction do
+          if profile.update(lead_provider_approval_status: "accepted") && other_profiles.update(lead_provider_approval_status: "rejected")
```

Comment:
Do we set papertrail whodunnit anywhere? Would be good to be able to work out how something got rejected

Reply:
Does papertrail work with `current_user` by default? If so, `ApiTokenAuthenticatable` should (I think) take care of it.

Reply:
`whodunnit` was not set for the api. I have added to `ApiTokenAuthenticatable`. Also the format of the `whodunnit` for the api is now a `GlobalID` thing

Reply:
reverted the paper trail commit and will go into another PR